### DISCUSSION TITLE
我想将SLANET表格识别模型导出成静态模型

### DISCUSSION THREAD

**Discussion by Tyx-main** [PaddlePaddle/PaddleOCR#14446] [Q&A]
我想将SLANET表格识别模型导出成静态模型，采用的网络结构是[TableAttentionHead](https://github.com/PaddlePaddle/PaddleOCR/blob/main/ppocr/modeling/heads/table_att_head.py)，以下是我的导出代码：
```
  import onnx
    # 实例化模型
    model = TableAttentionHead(
        in_channels=[3],  
        hidden_size=256,         
        out_channels=30,         
    )
    
    # 指定输入
    input_spec = [paddle.static.InputSpec(shape=[1, 3, 488, 488], dtype='float32')]
    
    # 将模型转换为静态图
    model = paddle.jit.to_static(model, input_spec=input_spec)
    # 保存为静态图模型
    paddle.jit.save(model, "static_model")
```

导出模型后发现只导出了static_model.json和static_model.pdiparams，并没有看到pdmodel，请问是我哪里操作不对吗？

**Comment by GreatV** [comment]


**Reply by Tyx-main** [reply]
# 实例化完整的 SLANet 模型
model = SLANet(config, **kwargs)

可以提供一个config的参考值吗

**Comment by GreatV** [comment]
json格式是新IR格式，如果想得到旧的pdmodel可以 export FLAGS_enable_pir_api=0，后重新导一遍。

**Comment by Tyx-main** [comment]
我尝试按照你说的使用from ppocr.modeling.architectures import SLANet 导出一个表格模型
但是有报错，似乎是没有找到SLANet的网络结构：
Traceback (most recent call last):
  File "/app/test/test_transfer_onnx.py", line 535, in <module>
    main()
  File "/app/test/test_transfer_onnx.py", line 497, in main
    from PaddleOCR.ppocr.modeling.architectures import SLANet
ImportError: cannot import name 'SLANet' from 'PaddleOCR.ppocr.modeling.architectures' (/app/PaddleOCR/ppocr/modeling/architectures/__init__.py)


以下是我的代码：
    import onnx
    from PaddleOCR.ppocr.modeling.architectures import SLANet
    model = SLANet() 
    # 指定输入
    input_spec = [paddle.static.InputSpec(shape=[1, 3, 488, 488], dtype='float32')]

    # 转换为静态图
    model = paddle.jit.to_static(model, input_spec=input_spec)

    # 保存为静态图模型
    paddle.jit.save(model, "static_model", combine_params=True)

**Comment by GreatV** [comment]
之前是机器人回复的，不一定正确，请按照第二次回复的尝试。
